aiokafka/conn.py

```diff
@@ -369,6 +403,69 @@ async def _do_sasl_handshake(self):
                 self._sasl_mechanism,
             )
 
+    def _create_sasl_reauthentication_task(
+        self, session_lifetime_ms: int
+    ) -> asyncio.Task:
+        self_ref = weakref.ref(self)
+        timeout = calculate_sasl_reauthentication_time(session_lifetime_ms)
+
+        log.info(
+            "SASL re-authentication required after %ds for connection %s:%s",
+            timeout,
+            self._host,
+            self._port,
+        )
+
+        sasl_reauthentication_task = create_task(
+            self._sasl_reauthentication(self_ref, timeout)
+        )
+        sasl_reauthentication_task.add_done_callback(
+            functools.partial(self._on_sasl_reauthentication_task_error, self_ref)
+        )
+
+        return sasl_reauthentication_task
+
+    @staticmethod
+    async def _sasl_reauthentication(
+        self_ref: weakref.ReferenceType["AIOKafkaConnection"],
+        sasl_reauthentication_time: int,
+    ) -> None:
+        self = self_ref()
```

Comment:
The weak reference pattern is defeated by acquiring `self` before the sleep. Consider moving `await asyncio.sleep()` before getting a strong ref to self. Or explain why we should keep the strong ref while sleeping